temba/api/v2/views.py

```diff
@@ -5736,3 +5741,80 @@ def get_read_explorer(cls):
                 query="export_csv=false",
             ),
         )
+
+
+class TwilioPhoneValidationEndpoint(BaseAPIView):
+    """
+    This endpoint provides a way to retrieve additional information about a phone number
+
+    A **GET** returns information about the specified phone number
+
+    * **phone_number** - the phone number you are requesting information about
+
+    Example:
+
+        GET /api/v2/twilio_phone_validation.json
+        {
+            "phone_number": "+15108675310",
+        }
+
+    Response:
+
+        {
+            "caller_name": null,
+            "carrier": {
+                "error_code": null,
+                "mobile_country_code": "310",
+                "mobile_network_code": "456",
+                "name": "verizon",
+                "type": "mobile"
+            },
+            "country_code": "US",
+            "national_format": "(510) 867-5310",
+            "phone_number": "+15108675310",
+            "add_ons": null,
+            "url": "https://lookups.twilio.com/v1/PhoneNumbers/+15108675310"
+        }
+    """
+
+    permission = "orgs.org_api"
+
+    def get(self, request, *args, **kwargs):
+        org = self.request.user.get_org()
+        client = None if org is None else org.get_twilio_client()
+        error_response = Response(
+            {"error": "Your organization is not connected to a Twilio account or the credentials are expired."},
+            status=status.HTTP_401_UNAUTHORIZED,
+        )
+        if not client:
+            return error_response
+
+        params = {key: request.data.get(key) for key in request.data}
+        params.update({key: request.query_params.get(key) for key in request.query_params})
+        serializer = TwilioNumberValidationSerializer(data=params)
+        if not serializer.is_valid():
+            return Response({"error": serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            response = client.lookups.v1.phone_numbers(serializer.validated_data["phone_number"]).fetch(
+                type=["carrier"]
+            )
+            return Response(response._properties, status=status.HTTP_200_OK)
+        except TwilioRestException:
```

Comment:
returning `error_response` with `status.HTTP_401_UNAUTHORIZED` when there is `TwilioRestException` can be really misleading. Like when I requested with a valid number format but not existing number, it is still telling me `Your organization is not connected to a Twilio account or the credentials are expired.` when error status from twilio is 404 with the following error message `"Unable to fetch record: The requested resource /PhoneNumbers/23480678865658 was not found"`

```suggestion
    except TwilioRestException as e:
            return Response({"error": e.msg}, status=e.status)
```

Reply:
@musamusa, I have added requested changes